docs/source/torchao_hf_integration.md

```diff
@@ -0,0 +1,380 @@
+
+# Save tokenizer
+tokenizer.push_to_hub(save_to)
+```
+
+```{seealso}
+For more information on quantization configs, see {class}`torchao.quantization.Int4WeightOnlyConfig`, {class}`torchao.quantization.Float8DynamicActivationInt4WeightConfig`, and {class}`torchao.quantization.Int8WeightOnlyConfig`.
+```
+
+```{note}
+For more information on supported quantization and sparsity configurations, see [HF-Torchao Docs](https://huggingface.co/docs/transformers/main/en/quantization/torchao).
+```
+
+(serving-with-vllm)=
+### 2. Serving with VLLM
+
+```{note}
+For more information on serving and inference with VLLM, please refer to [Integration with VLLM: Architecture and Usage Guide](https://docs.pytorch.org/ao/main/torchao_vllm_integration.html) and [(Part 3) Serving on vLLM, SGLang, ExecuTorch](https://docs.pytorch.org/ao/main/serving.html) for a full end-to-end tutorial.
+```
+
+(Inference-with-HuggingFace-Transformers)=
+### 3. Inference with HuggingFace Transformers
+
+Recall how we can quantize models using HuggingFace Transformers.
+
+```python
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+torch.random.manual_seed(0)
+
+model_path = "pytorch/Phi-4-mini-instruct-float8dq"
+
+# Load and automatically quantize the model
+model = AutoModelForCausalLM.from_pretrained(
+    model_path,
+    device_map="auto",
+    torch_dtype="auto",
+    trust_remote_code=True,
+)
+tokenizer = AutoTokenizer.from_pretrained(model_path)
+```
+
+Now we can use the model for inference.
+
+```python
+from transformers import pipeline
+
+# Simulate conversation between user and assistant
+messages = [
+    {"role": "system", "content": "You are a helpful AI assistant."},
+    {"role": "user", "content": "Can you provide ways to eat combinations of bananas and dragonfruits?"},
+    {"role": "assistant", "content": "Sure! Here are some ways to eat bananas and dragonfruits together: 1. Banana and dragonfruit smoothie: Blend bananas and dragonfruits together with some milk and honey. 2. Banana and dragonfruit salad: Mix sliced bananas and dragonfruits together with some lemon juice and honey."},
+    {"role": "user", "content": "What about solving an 2x + 3 = 7 equation?"},
+]
+
+# Initialize HuggingFace pipeline for text generation
+pipe = pipeline(
+    "text-generation",
+    model=model,
+    tokenizer=tokenizer,
+)
+
+generation_args = {
+    "max_new_tokens": 500,
+    "return_full_text": False,
+    "temperature": 0.0,
+    "do_sample": False,
+}
+
+# Generate output
+output = pipe(messages, **generation_args)
+print(output[0]['generated_text'])
+```
+
+```{seealso}
+For more examples and recommended quantization methods based on different hardwares (i.e. A100 GPU, H100 GPU, CPU), see [HF-Torchao Docs (Quantization Examples)](https://huggingface.co/docs/transformers/main/en/quantization/torchao#quantization-examples).
+```
+
+(Inference-with-HuggingFace-Diffuser)=
+### 4. Inference with HuggingFace Diffuser
```

Comment:
Right now this reads like the 4th step after "Step 3: Inference with HuggingFace Transformers". Should it be like 3a and 3b instead?